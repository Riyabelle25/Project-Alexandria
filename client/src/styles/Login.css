.login {
  display: grid;
  place-items: center;
  height: 100%;
  width: 100%;
}

.login > button {
  width: 300px;
  background-color: bisque;
  color: black;
  font-weight: 800;
}

.login > button:hover {
  background-color: white;
  color:brown;
}

.login > div > img {
  object-fit: contain;
  height: 150px;

}
.App-logo {
  height: 40vmin;
  width: 40vmin;
  pointer-events: none;
  padding: 5%;
}
/* .MuiButton-root,.MuiButton-label{
  width: 10%;
  height: 7vh;
  background-color: rgb(207, 207, 149);
  color: rgb(34, 2, 2);
  border-radius: 17%;
  font-size: medium;
} */
#welcome-msg{
  color: rgb(34, 2, 2);
}
@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-bounce 4s ease 0s infinite;
    /* name duration timing-function delay iteration-count direction fill-mode; */
  }
}
@keyframes App-logo-bounce {
  0%   { transform: translateY(0); }
        50%  { transform: translateY(50px); }
        100% { transform: translateY(0); }
}
