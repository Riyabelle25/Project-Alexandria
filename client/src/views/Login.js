import React from 'react';
import { Column, Row, Item } from '@mui-treasury/components/flex';
import { Button } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import wall from '../texture-wall.jpeg';
import { auth, provider } from '../app/firebase';
import '../styles/Login.css';
import logo from '../alexandria.svg'
import { urlAlphabet } from 'nanoid';

function Login() {
  const signIn = () => {
    auth.signInWithPopup(provider).catch((error) => alert(error.message));
  };

  return (
    <div className="login">
     <header className="App-header">
       <Paper id="paper" elevation={7} style={{ border:"6px solid #c49c48", padding:"6%", alignContent:"center",backgroundImage:`url(${wall})`, borderRadius:"0.5rem"}}>
       <h1 id="welcome-msg">Welcome to Alexandria!</h1>
      <img src={logo} className="App-logo" alt="logo" style={{marginLeft:"12%"}} />
      <Row justifyContent={"center"} marginTop={"6%"} >
      <Item >
      <button onClick={signIn} id="login-btn" style={{border: "1 px solid #c49c48", fontFamily:'Lucida Grande',
        fontSize:"80%", color:"rgb(37, 5, 63)",backgroundColor:"#bcaaa4", borderRadius:"1rem", width: "7vw", height: "6vh"}}>Sign In</button>
      </Item>
      <Item paddingLeft={4} paddingTop={1} fontFamily="Lucida Grande"><h3>or </h3></Item>
      <Item>
        <button id="video-call" onClick={()=> window.location.href=`/meeting/${Math.floor(Math.random()* 33)*1000}`} 
        style={{border: "1 px solid #c49c48", fontFamily:'Lucida Grande',
        fontSize:"80%", color:"rgb(37, 5, 63)",backgroundColor:"#bcaaa4", borderRadius:"1rem", width: "14vw", height: "6vh"}}>
          Start a Video Call
        </button>
      </Item>
      </Row>
      
      </Paper>
      </header>
    </div>
  );
}

export default Login;
