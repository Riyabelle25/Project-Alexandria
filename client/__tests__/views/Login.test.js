import Login from "../../src/views/Login";
import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import toJson from "enzyme-to-json";
import { Provider } from "react-redux";

import configureStore from "redux-mock-store"; // Smart components

configure({ adapter: new Adapter() });

describe("<Login />", () => {
  const initialState = { user: null };
  const mockStore = configureStore();
  let store;
  describe("render()", () => {
    test("renders the component", () => {
      store = mockStore(initialState);
      const wrapper = shallow(
        <Provider store={store}>
          <Login />
        </Provider>
      );
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
