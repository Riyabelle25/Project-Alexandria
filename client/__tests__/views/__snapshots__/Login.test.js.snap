// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> render() renders the component 1`] = `<Login />`;
